### PR TITLE
GH-100719: Remove the `co_nplaincellvars` field from code objects.

### PR DESCRIPTION
**Pull request by markshannon**
Removes a redundant field

* Issue: gh-100719

### CODE REVIEW

File: Objects/typeobject.c [hunk 1]
```diff
@@ -9514,7 +9514,7 @@ super_init_without_args(_PyInterpreterFrame *cframe, PyCodeObject *co,
 
     // Look for __class__ in the free vars.
     PyTypeObject *type = NULL;
-    int i = co->co_nlocals + co->co_nplaincellvars;
+    int i = PyCode_GetFirstFree(co);
```
Comment (iritkatriel) [inline Objects/typeobject.c:9517]: Py_ssize_t?

File: Objects/frameobject.c [hunk 2]
```diff
@@ -1119,7 +1119,7 @@ frame_init_get_vars(_PyInterpreterFrame *frame)
 
     /* Free vars have not been initialized -- Do that */
     PyObject *closure = ((PyFunctionObject *)frame->f_funcobj)->func_closure;
-    int offset = co->co_nlocals + co->co_nplaincellvars;
+    int offset = PyCode_GetFirstFree(co);
```
Comment (iritkatriel) [inline Objects/frameobject.c:1122]: Py_ssize_t?

File: Python/compile.c [hunk 3]
```diff
@@ -2260,7 +2260,7 @@ compiler_make_closure(struct compiler *c, location loc,
         qualname = co->co_name;
 
     if (co->co_nfreevars) {
-        int i = co->co_nlocals + co->co_nplaincellvars;
+        int i = PyCode_GetFirstFree(co);
```
Comment (iritkatriel) [inline Python/compile.c:2263]: Py_ssize_t?

File: Objects/typeobject.c (hunk 1 above)
Comment (markshannon) [reply Objects/typeobject.c:9517]: This should be an int. I'll change `PyCode_GetFirstFree`.

**Review by iritkatriel — APPROVED** [review]

File: Objects/typeobject.c (hunk 1 above)
Comment (obataku) [reply Objects/typeobject.c:9517]: @markshannon why should it be `int`?

**Pull request merged**